src/test/codegen/vec-in-place.rs

```diff
@@ -29,44 +29,47 @@ pub struct Bar {
 // CHECK-LABEL: @vec_iterator_cast_primitive
 #[no_mangle]
 pub fn vec_iterator_cast_primitive(vec: Vec<i8>) -> Vec<u8> {
-    // CHECK-NOT: loop
```

Comment:
The dropped `CHECK-NOT: loop` looks suspicious. What does the generated IR look like now?

Reply:
I have a copy here. https://gist.github.com/cjgillot/b29ba3c9be80074111f163f4de4476cf

Reply:
Thanks. It looks like the first three cases still optimize fine, but vec_iterator_cast_aggregate and vec_iterator_cast_deaggregate produce exactly the kind of code we don't want (a runtime unrolled loop rather than a fixed number of loads and stores).

Most of these tests were added recently by https://github.com/rust-lang/rust/pull/95695, cc @the8472.

Reply:
Yeah, the test exists to highlight accidental regressions of a nice-to-have optimization, it's not part of std guarantees or anything like that.

I don't understand the changes from

`// CHECK-NOT: call`

to

`// CHECK-NOT: @__rust_alloc`

The original intent of those checks was to make sure that there is neither loops nor calls to functions which might contain loops. The IR looked fairly simple, as it still does for `@vec_iterator_cast_primitive`. It wasn't about merely checking the absence of allocations.

Reply:
Because I missed a piece when reverting changes to the test.